src/mechanics/tools/QuasistaticSolver.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "mechanics/tools/TimeDependentProblem.h"
+#include "mechanics/constraints/Constraints.h"
+#include "boost/optional.hpp"
+
+namespace NuTo
+{
+class QuasistaticSolver
+{
+public:
+    //! Ctor
+    //! @param equations system of equations including Gradient(), Hessian0() and UpdateHistory()
+    //! @param dof dof type
+    QuasistaticSolver(TimeDependentProblem& equations, DofType dof);
+
+    //! @param constraints linear constraints
+    void SetConstraints(Constraint::Constraints constraints);
+
+    //! sets the global time required for evaluating the constraint right hand side
+    //! @param globalTime global time
+    void SetGlobalTime(double globalTime);
```

Comment:
Shouldn't the methods that change the state be private (Update, SetTime)? They should IMO only be called by the DoStep method. Similarly, I'm not yet sure how to interpret the setting of mX in this context (not yet implemented due to the assumption of zero initial conditions). On the one hand, this also should be private, but this would not allow to set initial conditions.

Reply:
I also think it would be a good idea to split the classes (not required now in this pull request) between the solver and the problem. So the DoStep method of the solver should be called with a specific problem.

Reply:
I kept the methods to manually implement my custom `DoStep` that, for example, does not use an newton raphson algorithm to find the equilibrium. Or one that allows further configuration of the solver. But you are right, calling `SetGlobalTime()` in between `DoStep` will mess things up and has to be avoided. I'll think about that.